Source/WebCore/style/values/color/StyleDynamicRangeLimit.cpp

```diff
@@ -0,0 +1,112 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "StyleDynamicRangeLimit.h"
+
+#include "AnimationUtilities.h"
+#include "CSSDynamicRangeLimit.h"
+#include "CSSDynamicRangeLimitMix.h"
+#include "StyleDynamicRangeLimitMix.h"
+#include <wtf/text/TextStream.h>
+
+namespace WebCore {
+namespace Style {
+
+using namespace CSS::Literals;
+
+// Resolves a `dynamic-range-limit-mix` function value for the `dynamic-range-limit` property.
+static DynamicRangeLimit resolve(DynamicRangeLimitMixFunction&& mix)
+{
+    // https://drafts.csswg.org/css-color-hdr/#computing-dynamic-range-limit
+
+    // This function implements steps 5 and 6 of the algorithm for computing the dynamic
+    // range limit value. The first 4 steps are in `ToStyle<CSS::DynamicRangeLimitMixFunction>`.
+
+    if (mix->standard == 100_css_percentage)
+        return DynamicRangeLimit(CSS::Keyword::Standard { });
+    if (mix->constrainedHigh == 100_css_percentage)
+        return DynamicRangeLimit(CSS::Keyword::ConstrainedHigh { });
+    if (mix->high == 100_css_percentage)
+        return DynamicRangeLimit(CSS::Keyword::High { });
+    return DynamicRangeLimit(WTFMove(mix));
+}
+
+// MARK: - Conversion
+
+auto ToCSS<DynamicRangeLimit>::operator()(const DynamicRangeLimit& limit, const RenderStyle& style) -> CSS::DynamicRangeLimit
+{
+    return WTF::switchOn(limit,
+        [&](const auto& value) -> CSS::DynamicRangeLimit {
+            return CSS::DynamicRangeLimit(toCSS(value, style));
+        }
+    );
+}
+
+auto ToStyle<CSS::DynamicRangeLimit>::operator()(const CSS::DynamicRangeLimit& limit, const BuilderState& state) -> DynamicRangeLimit
+{
+    return WTF::switchOn(limit,
+        [&]<CSSValueID Id>(const Constant<Id>& keyword) -> DynamicRangeLimit {
+            return DynamicRangeLimit(keyword);
+        },
+        [&](const CSS::DynamicRangeLimitMixFunction& mix) -> DynamicRangeLimit {
+            return resolve(toStyle(mix, state));
+        }
+    );
+}
+
+// MARK: - Blending
+
+auto Blending<DynamicRangeLimit>::blend(const DynamicRangeLimit& from, const DynamicRangeLimit& to, const BlendingContext& context) -> DynamicRangeLimit
+{
+    // Blending is defined as "by dynamic-range-limit-mix()". This requires computing the equivalent of:
+    //
+    //   dynamic-range-limit-mix(`from` (1 - context.progress) * 100%, `to` context.progress * 100%)
+    //
+    // and then resolving that as we would when converting from CSS::DynamicRangeLimit. Rather than
+    // construct a full fledged CSS::DynamicRangeLimitMixFunction, we can utilize the weighted sum
+    // addition functionality to construct just what we need and then resolve that directly.
+
+    DynamicRangeLimitMixPercentage fromMixPercentage = (1.0 - context.progress) * 100.0;
+    DynamicRangeLimitMixPercentage toMixPercentage   = (context.progress) * 100.0;
+
+    DynamicRangeLimitMixFunction function { .parameters = { 0_css_percentage, 0_css_percentage, 0_css_percentage } };
```

Comment:
Can `auto` be used here?

Reply:
I don’t think so. How are you imagining that?